src/collections/Breadcrumb/Breadcrumb.js

```diff
@@ -0,0 +1,106 @@
+import _ from 'lodash'
+import cx from 'classnames'
+import React, { PropTypes } from 'react'
+import META from 'src/utils/Meta'
+import { customPropTypes } from '../../utils/propUtils'
+import * as sui from '../../utils/semanticUtils'
+import BreadcrumbDivider from './BreadcrumbDivider'
+import BreadcrumbSection from './BreadcrumbSection'
+
+/**
+ * A breadcrumb is used to show hierarchy between content.
+ */
+function Breadcrumb(props) {
+  const {
+    children, divider, icon, size, sections, ...rest,
+  } = props
+
+  const classes = cx(
+    'ui',
+    size,
+    'breadcrumb'
+  )
+
+  if (!sections) {
+    return <div className={classes} {...rest}>{children}</div>
+  }
+
+  let dividerJSX = <Breadcrumb.Divider />
+
+  if (divider) {
+    dividerJSX = <Breadcrumb.Divider>{divider}</Breadcrumb.Divider>
+  }
+
+  if (icon) {
+    dividerJSX = <Breadcrumb.Divider icon={icon} />
+  }
+
+  return (
+    <div className={classes} {...rest}>
+      {sections.map((section, index) => {
+        const isLast = index === (sections.length - 1)
+
+        return [
+          <Breadcrumb.Section {...section}>{section.text}</Breadcrumb.Section>,
+          isLast ? null : dividerJSX,
+        ]
+      })}
+    </div>
+  )
```

Comment:
Regarding the `render()` function:

- prefer to pull js expressions into variables or smaller render methods, i.e. `renderSections()`.
- children in an array need a unique `key`, see [dynamic children](https://facebook.github.io/react/docs/multiple-components.html#dynamic-children)
- `children` here is a 2 dimensional array with a the last nested child being `null`, this could be a flat array with no null values.
- spreading section `{ ...section }` is also going to spread props we don't need, like `text`.  suggest destructuring `text` and spreading the `...rest`.

Some refactoring to address the above may look something like this:

```jsx
const sectionsJSX = []

sections.forEach(({ text, ...restSection }, index) => {
  const key = `${text}-${index}`

  sectionsJSX.push(
    <Breadcrumb.Section {...restSection} key={key}>{text}</Breadcrumb.Section>
  )

  if (index !== sections.length - 1) {
    sectionsJSX.push(React.cloneElement(dividerJSX, { key }))
  }
})

return <div className={classes} {...rest}>{sectionsJSX}</div>
```